newrelic/resource_newrelic_api_access_key_test.go

```diff
@@ -0,0 +1,162 @@
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckNewRelicAPIAccessKeyUser(accountID, userID, keyName, keyNotes),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("newrelic_api_access_key.foobar", "account_id", accountIDRaw),
+					resource.TestCheckResourceAttr("newrelic_api_access_key.foobar", "key_type", KeyTypeUser),
+					resource.TestCheckResourceAttr("newrelic_api_access_key.foobar", "user_id", userIDRaw),
+					resource.TestCheckResourceAttr("newrelic_api_access_key.foobar", "name", keyName),
+					resource.TestCheckResourceAttr("newrelic_api_access_key.foobar", "notes", keyNotes),
+					resource.TestCheckResourceAttrSet("newrelic_api_access_key.foobar", "key"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccNewRelicAPIAccessKey_BasicIngestBrowserNoNotesNames(t *testing.T) {
+	accountIDRaw, accountID := retrieveIdsFromEnvOrSkip(t, "NEW_RELIC_TEST_ACCOUNT_ID")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckNewRelicAPIAccessKeyIngestNoNameNotes(accountID, KeyTypeIngestBrowser),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("newrelic_api_access_key.foobar", "account_id", accountIDRaw),
+					resource.TestCheckResourceAttr("newrelic_api_access_key.foobar", "key_type", KeyTypeIngest),
+					resource.TestCheckResourceAttr("newrelic_api_access_key.foobar", "ingest_type", KeyTypeIngestBrowser),
+					resource.TestCheckResourceAttr("newrelic_api_access_key.foobar", "name", ""),
+					resource.TestCheckResourceAttr("newrelic_api_access_key.foobar", "notes", ""),
+					resource.TestCheckResourceAttrSet("newrelic_api_access_key.foobar", "key"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckNewRelicAPIAccessKeyIngestNoNameNotes(accountID int, ingestType string) string {
+	return fmt.Sprintf(`
+resource "newrelic_api_access_key" "foobar" {
+	account_id  = %d
+	key_type    = "INGEST"
+	ingest_type = "%s"
+}
+`, accountID, ingestType)
+}
+
+func testAccCheckNewRelicAPIAccessKeyIngest(accountID int, ingestType, name, notes string) string {
+	return fmt.Sprintf(`
+resource "newrelic_api_access_key" "foobar" {
+	account_id  = %d
+	key_type    = "INGEST"
+	ingest_type = "%s"
+	name        = "%s"
+    notes       = "%s"
+}
+`, accountID, ingestType, name, notes)
+}
+
+func testAccCheckNewRelicAPIAccessKeyUser(accountID, userID int, name, notes string) string {
+	return fmt.Sprintf(`
+resource "newrelic_api_access_key" "foobar" {
+	account_id  = %d
+	key_type    = "USER"
+	user_id     = "%d"
+	name        = "%s"
+    notes       = "%s"
+}
+`, accountID, userID, name, notes)
+}
+
+// retrieveIdsFromEnvOrSkip retrieves certain variables from the environment needed by the api access key tests.
+func retrieveIdsFromEnvOrSkip(t *testing.T, envKey string) (string, int) {
```

Comment:
Maybe this method could move to `helpers_test.go`.  It seems useful for more than just this test case.